editor/src/components/canvas/controls/grid-controls.tsx

```diff
@@ -2163,10 +2175,269 @@ function useIsGridItemInteractionActive() {
+      )
+      if (cellBounds == null) {
+        return null
+      }
+
+      const cellBoundsColumnIndex = cellBounds.column - 1
+      const firstRow = parentGridCellGlobalFrames[0]
+      const left = firstRow[cellBoundsColumnIndex].x
+      const width = getCellCanvasWidthFromBounds(
+        parentGridCellGlobalFrames,
+        cellBoundsColumnIndex,
+        cellBounds.width,
+      )
+
+      const cellBoundsRowIndex = cellBounds.row - 1
+      const firstColumn = parentGridCellGlobalFrames[cellBoundsRowIndex][0]
+      const top = firstColumn.y
+      const height = getCellCanvasHeightFromBounds(
+        parentGridCellGlobalFrames,
+        cellBoundsRowIndex,
+        cellBounds.height,
+      )
+
+      const gridRect = MetadataUtils.getFrameOrZeroRectInCanvasCoords(
+        originalGrid,
+        store.editor.jsxMetadata,
+      )
+
+      return {
+        columnStart: {
+          top: gridRect.y,
+          left: left,
+          position: elementGridProperties.gridColumnStart,
+          counterpart: elementGridProperties.gridColumnEnd,
+          bound: 'start',
+        },
+        columnEnd: {
+          top: gridRect.y,
+          left: left + width,
+          position: elementGridProperties.gridColumnEnd,
+          counterpart: elementGridProperties.gridColumnStart,
+          bound: 'end',
+        },
+        rowStart: {
+          top: top,
+          left: gridRect.x,
+          position: elementGridProperties.gridRowStart,
+          counterpart: elementGridProperties.gridRowEnd,
+          bound: 'start',
+        },
+        rowEnd: {
+          top: top + height,
+          left: gridRect.x,
+          position: elementGridProperties.gridRowEnd,
+          counterpart: elementGridProperties.gridRowStart,
+          bound: 'end',
+        },
+      }
+    },
+    'RulerMarkers markers',
+  )
+
+  if (markers == null) {
+    return null
+  }
+
+  return (
+    <React.Fragment>
+      <RulerMarkerIndicator marker={markers.columnStart} axis={'column'} />
+      <RulerMarkerIndicator marker={markers.columnEnd} axis={'column'} />
+      <RulerMarkerIndicator marker={markers.rowStart} axis={'row'} />
+      <RulerMarkerIndicator marker={markers.rowEnd} axis={'row'} />
+    </React.Fragment>
+  )
+})
+RulerMarkers.displayName = 'RulerMarkers'
+
+const RulerMarkerIndicator = React.memo(
+  (props: { marker: RulerMarkerPositionData; axis: 'row' | 'column' }) => {
+    const colorTheme = useColorTheme()
+
+    const markerType = getRulerMarkerType({
+      position: props.marker.position,
+      counterpart: props.marker.counterpart,
+      bound: props.marker.bound,
+    })
+    const markerIcon = rulerMarkerIcons[markerType][props.axis]
+
+    const canvasScale = useEditorState(
+      Substores.canvasOffset,
+      (store) => store.editor.canvas.scale,
+      'RulerMarkerIndicator canvasScale',
+    )
+
+    function skewMarkerPosition(axis: 'column' | 'row') {
+      return props.axis === axis
```

Comment:
Minor: I think this would be clearer as a couple of if statements.

Reply:
[`e26a3b0` (#6654)](https://github.com/concrete-utopia/utopia/pull/6654/commits/e26a3b033aaaa007fcad493de973dee00780d76c)